lib/commands/server_handshake.js

```diff
@@ -75,6 +76,90 @@ class ServerHandshake extends Command {
     const encoding = connection.clientHelloReply.encoding;
     const commandCode = packet.readInt8();
     switch (commandCode) {
+      case CommandCode.STMT_PREPARE:
+        if (connection.listeners('stmt_prepare').length) {
+          const query = packet.readString(undefined, encoding);
+          connection.emit('stmt_prepare', query);
+        } else {
+          connection.writeError({
+            code: Errors.HA_ERR_INTERNAL_ERROR,
+            message:
+              'No query handler for prepared statements.'
+          });
+        }
+        break;
+      case CommandCode.STMT_EXECUTE:
+        if (connection.listeners('stmt_execute').length) {
+          console.log(packet)
+          const stmtId = packet.readInt32();
```

Comment:
this should be moved to `PrepareStatement.fromPacket`, added [here](https://github.com/sidorares/node-mysql2/blob/5022b54b012872d062b260a0cafb2a134139dd30/lib/packets/prepare_statement.js#L24) similar to https://github.com/sidorares/node-mysql2/blob/5022b54b012872d062b260a0cafb2a134139dd30/lib/packets/prepare_statement.js#L24

Reply:
This was added to `/lib/packets/execute.js`. Hopefully this is good enough.

Reply:
thanks! ( yep, meant execute packet, not prepare )